### PR TITLE
update ibm k8s agent scripts - add local image analyzer

### PR DESCRIPTION
**Pull request by mbertrone**
Update agent install script used in IBM environment, including the scanning local image analyzer.

### CODE REVIEW

File: agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh [hunk 1]
```diff
@@ -245,6 +260,39 @@ function install_k8s_agent {
...
+      fi
+      if [ ! -z "$DOCKER_SOCKET_PATH" ]; then
+        echo "* Setting docker socket path"
+        echo "  docker_socket_path: $DOCKER_SOCKET_PATH" >> $IA_CONFIG_FILE
+      fi
+      if [ ! -z "$CRI_SOCKET_PATH" ]; then
+        echo "* Setting CRI socket path"
+        echo "  cri_socket_path: $CRI_SOCKET_PATH" >> $IA_CONFIG_FILE
+      fi
+      if [ ! -z "$CHECK_CERT" ]; then
```
Comment (mbreitung) [inline agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh:286]: You should be able to remove the `else` here by defaulting CHECK_CERT. I don't think that the `else` here is doing anything.

File: agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh [hunk 2]
```diff
@@ -245,6 +260,39 @@ function install_k8s_agent {
...
+        echo "* Setting SSL certificate check level"
+        echo "  ssl_verify_certificate: \"$CHECK_CERT\"" >> $IA_CONFIG_FILE
```
Comment (mbreitung) [inline agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh:288]: why the quotes around the value?

File: agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh [hunk 3]
```diff
@@ -308,6 +356,40 @@ function install_k8s_agent {
 
     echo "* Deploying the sysdig agent"
     kubectl apply -f $DAEMONSET_FILE --namespace=$NAMESPACE
+
+    echo -e "\nThe list of agent pods deployed in the namespace \"$NAMESPACE\" are:"
```
Comment (mbreitung) [inline agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh:360]: the `-e` here can be refactored out. It is not needed

File: agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh [hunk 4]
```diff
@@ -308,6 +356,40 @@ function install_k8s_agent {
...
+    kubectl get pods -n $NAMESPACE | grep "sysdig-agent"
+
+    echo -e "\nMake sure the above pods all turn to \"Running\" state before continuing"
+    echo "Should any pod not reach the \"Running\" state, further info can be obtained from logs as follows"
+    echo "'kubectl logs <agent-pod-name> -n $NAMESPACE' "
+
+    if [ $INSTALL_ANALYZER -eq 1 ]; then
+      # Deploy Image Analyzer
+      IA_FILE=/tmp/sysdig-image-analyzer-daemonset.yaml
```
Comment (mbreitung) [inline agent_deploy/IBMCloud-Kubernetes-Service/install-agent-k8s.sh:369]: please expand variable names. I assume that IA==IMAGE_ANALYZER?

**Review by luca-sd — APPROVED** [review]
We have updated and tested this against a dev IBM cluster today

**Comment by luca-sd** [comment]
@mbreitung re. Bash style, we'll need to keep this in sync with our Agent repo. While this is not ideal, if we change style here we won't be able to easily port modifications from the other repository.

**Review by mbreitung — APPROVED** [review]

**Pull request merged**